src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/EnclaveDelegate.NetCoreApp.cs

```diff
@@ -82,5 +86,119 @@ internal byte[] GetSerializedAttestationParameters(SqlEnclaveAttestationParamete
                 if (sqlEnclaveSession == null) throw SQL.NullEnclaveSessionReturnedFromProvider(enclaveType, attestationUrl);
             }
         }
+
+        /// <summary>
+        /// Generate the byte package that needs to be sent to the enclave
+        /// </summary>
+        /// <param name="attestationProtocol">attestation protocol</param>
+        /// <param name="keysTobeSentToEnclave">Keys to be sent to enclave</param>
+        /// <param name="queryText"></param>
+        /// <param name="enclaveType">enclave type</param>
+        /// <param name="serverName">server name</param>
+        /// <param name="enclaveAttestationUrl">url for attestation endpoint</param>
+        /// <returns></returns>
+        internal EnclavePackage GenerateEnclavePackage(SqlConnectionAttestationProtocol attestationProtocol, Dictionary<int, SqlTceCipherInfoEntry> keysTobeSentToEnclave, string queryText, string enclaveType, string serverName, string enclaveAttestationUrl)
+        {
+
+            SqlEnclaveSession sqlEnclaveSession = null;
+            long counter;
+            try
+            {
+                GetEnclaveSession(attestationProtocol, enclaveType, serverName, enclaveAttestationUrl, out sqlEnclaveSession, out counter, throwIfNull: true);
+            }
+            catch (Exception e)
+            {
+                throw new RetriableEnclaveQueryExecutionException(e.Message, e);
+            }
+
+            List<ColumnEncryptionKeyInfo> decryptedKeysToBeSentToEnclave = GetDecryptedKeysToBeSentToEnclave(keysTobeSentToEnclave, serverName);
+            byte[] queryStringHashBytes = ComputeQueryStringHash(queryText);
+            byte[] keyBytePackage = GenerateBytePackageForKeys(counter, queryStringHashBytes, decryptedKeysToBeSentToEnclave);
+            byte[] sessionKey = sqlEnclaveSession.GetSessionKey();
+            byte[] encryptedBytePackage = EncryptBytePackage(keyBytePackage, sessionKey, serverName);
+            byte[] enclaveSessionHandle = BitConverter.GetBytes(sqlEnclaveSession.SessionId);
+            byte[] byteArrayToBeSentToEnclave = CombineByteArrays(new[] { enclaveSessionHandle, encryptedBytePackage });
+            return new EnclavePackage(byteArrayToBeSentToEnclave, sqlEnclaveSession);
+        }
+
+        internal void InvalidateEnclaveSession(SqlConnectionAttestationProtocol attestationProtocol, string enclaveType, string serverName, string EnclaveAttestationUrl, SqlEnclaveSession enclaveSession)
+        {
+            SqlColumnEncryptionEnclaveProvider sqlColumnEncryptionEnclaveProvider = GetEnclaveProvider(attestationProtocol, enclaveType);
+            sqlColumnEncryptionEnclaveProvider.InvalidateEnclaveSession(serverName, EnclaveAttestationUrl, enclaveSession);
+        }
+
+        
+        internal SqlEnclaveAttestationParameters GetAttestationParameters(SqlConnectionAttestationProtocol attestationProtocol, string enclaveType, string serverName, string enclaveAttestationUrl)
```

Comment:
`serverName` and `enclaveAttestationUrl` not used in this function?
```suggestion
        internal SqlEnclaveAttestationParameters GetAttestationParameters(SqlConnectionAttestationProtocol attestationProtocol, string enclaveType)
```